### PR TITLE
Document the interface using Doxygen/Natspec commands

### LINKED ISSUE #1: Add natspec comments.

### PR DESCRIPTION
**Pull request by axic**
Closes #1.

cc @chriseth

### CODE REVIEW

**Review by MrChico — APPROVED** [review]

File: deposit_contract.sol [hunk 1]
```diff
@@ -12,15 +13,25 @@ interface IDepositContract {
         bytes index
     );
 
+    /// @notice Submit a Phase 0 DepositData object.
```
Comment (chriseth) [inline deposit_contract.sol:18]: I think the comments have to be on the contract itself to be visible in the metadata. Docstring inheritance is still an open issue.

Comment (axic) [reply deposit_contract.sol:18]: I did check it as I was writing it (the only expected part is that events are ignored, see https://github.com/ethereum/solidity/issues/8776):

```json
User Documentation
{
  "methods":
  {
    "deposit(bytes,bytes,bytes,bytes32)":
    {
      "notice": "Submit a Phase 0 DepositData object."
    },
    "get_deposit_count()":
    {
      "notice": "Query the current deposit count."
    },
    "get_deposit_root()":
    {
      "notice": "Query the current deposit root hash."
    }
  }
}
```

```json
Developer Documentation
{
  "methods":
  {
    "deposit(bytes,bytes,bytes,bytes32)":
    {
      "params":
      {
        "deposit_data_root": "The SHA-256 hash of the SSZ-encoded DepositData object. Used as a protection against malformed input.",
        "pubkey": "A BLS12-381 public key.",
        "signature": "A BLS12-381 signature.",
        "withdrawal_credentials": "Commitment to a public key for withdrawals."
      }
    }
  }
}
```

One thing I didn't understood why `@return` is ignored, will open an issue (https://github.com/ethereum/solidity/issues/8905).

Comment (axic) [reply deposit_contract.sol:18]: >I think the comments have to be on the contract itself to be visible in the metadata. Docstring inheritance is still an open issue.

Ah, understood after re-reading. Yes, what I pasted above is on the interface and not the contract, so it won't be included in the metadata.

**Comment by axic** [comment]
@chriseth @MrChico please review again

**Comment by axic** [comment]
If it needs to be, in a separate PR the content can be moved off the interface.

File: deposit_contract.sol [hunk 2]
```diff
@@ -3,7 +3,10 @@
 pragma solidity ^0.6.0;
 
 // This interface is designed to be compatible with the Vyper version.
+/// @notice This is the Ethereum 2.0 deposit contract interface.
+/// For more information see https://github.com/ethereum/eth2.0-specs/blob/v0.11.2/specs/phase0/deposit-contract.md
```
Comment (axic) [inline deposit_contract.sol]: Is it a good idea to lock in a version here? :)

**Comment by chriseth** [comment]
The comments are still on the interface.

**Comment by axic** [comment]
> The comments are still on the interface.

https://github.com/axic/eth2-deposit-contract/pull/3#issuecomment-628100806

**Comment by axic** [comment]
If we move natspec from interface to contract then interfaces become somewhat irrelevant. Merging this in its current form with the hope https://github.com/ethereum/solidity/issues/8911 is solved.

**Review by MrChico — APPROVED** [review]

**Pull request merged**